xunit.runner.visualstudio.desktop/TestPlatformContext.cs

```diff
@@ -0,0 +1,22 @@
+namespace Xunit.Runner.VisualStudio.TestAdapter
+{
+    /// <summary>
+    /// Provides contextual information on a test run/discovery based on runsettings
+    /// or the invocation (execution, discovery).
+    /// </summary>
+    public struct TestPlatformContext
```

Comment:
In #100, support for checking context was added. In this PR, we are adding another context on *when to serialize ITestCase*. This object holds all such requirements from vstest for decisions during discovery and execution.